server/src/test/java/io/spine/server/procman/PmMessageDeliveryShould.java

```diff
@@ -21,145 +21,205 @@
-                                                   .iterator()
-                                                   .next();
-
-        delivery.deliverNow(projectId, postponedEvents.get(projectId));
-
-        final PmProjectStarted deliveredEventMsg = ReactingProjectWizard.getEventReceived();
-        assertNotNull(deliveredEventMsg);
-        assertEquals(Events.getMessage(event), deliveredEventMsg);
+    public void dispatch_commands_to_single_shard_in_multithreaded_env() throws
+                                                                         Exception {
+
+        final ParallelDispatcher<ProjectId, Command> dispatcher =
+                new ParallelDispatcher<ProjectId, Command>(
+                        42, 400, dispatchWaitTime()) {
+                    @Override
+                    protected ThreadStats<ProjectId> getStats() {
+                        return DeliveryPm.getStats();
+                    }
+
+                    @Override
+                    protected Command newMessage() {
+                        return createProject();
+                    }
+
+                    @Override
+                    protected void postToBus(BoundedContext context, Command command) {
+                        context.getCommandBus()
+                               .post(command, StreamObservers.<Ack>noOpObserver());
+                    }
+                };
+
+        dispatcher.dispatchMessagesTo(new SingleShardPmRepository());
     }
 
-
     @Test
-    public void postpone_rejections_dispatched_to_reactor_method() {
-        assertNull(ReactingProjectWizard.getEventReceived());
-
-        final Rejection rejection = cannotStartProject();
-        boundedContext.getRejectionBus()
-                      .post(rejection);
-
-        assertNull(ReactingProjectWizard.getRejectionReceived());
-
-        final RejectionEnvelope expectedEnvelope = RejectionEnvelope.of(rejection);
-        final PostponingRejectionDelivery delivery = repository.getRejectionEndpointDelivery();
-        final Map<ProjectId, RejectionEnvelope> postponedRejections =
-                delivery.getPostponedRejections();
-        assertTrue(postponedRejections.size() == 1 &&
-                           postponedRejections.containsValue(expectedEnvelope));
-
-        final ProjectId projectId = postponedRejections.keySet()
-                                                       .iterator()
-                                                       .next();
-
-        delivery.deliverNow(projectId, postponedRejections.get(projectId));
-
-        final PmCannotStartArchivedProject deliveredRejectionMsg =
-                ReactingProjectWizard.getRejectionReceived();
-        assertNotNull(deliveredRejectionMsg);
-        assertEquals(getMessage(rejection), deliveredRejectionMsg);
+    public void dispatch_events_to_single_shard_in_multithreaded_env() throws
+                                                                       Exception {
```

Comment:
If these methods do throw exceptions, can we have the declarations coming on new lines?

Reply:
Not sure what you mean, but I have removed a redundant new-lines.